Lib/test/test_capi/test_watchers.py

```diff
@@ -336,6 +336,102 @@ def test_no_more_ids_available(self):
                 self.add_watcher()
 
 
+class TestCodeObjectWatchers(unittest.TestCase):
+    @contextmanager
+    def code_watcher(self, which_watcher):
+        wid = _testcapi.register_code_watcher(which_watcher)
+        try:
+            yield wid
+        finally:
+            _testcapi.clear_code_watcher(wid)
+
+    def make_counts_dict(self, created_1, destroyed_1, created_2, destroyed_2):
+        return {
+            0: {"created": created_1, "destroyed": destroyed_1},
+            1: {"created": created_2, "destroyed": destroyed_2},
+        }
+
+    def test_code_object_events_dispatched(self):
```

Comment:
nit: personally I would prefer this as ~5 separate test methods rather than a single long narrative test method (basically a test method for every header comment). It means you get more fine-grained single on failure about what case exactly failed. But I don't feel super strongly about it and I don't know if upstream will care.

Reply:
this isn't really possible, as this is testing a "stateful progression" between the two watchers